gems/sorbet-runtime/lib/types/private/final.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+# typed: false
+
+module T::Private::Final
+  def self.declare(mod)
+    if !mod.is_a?(Module)
+      raise "#{mod.name} is not a module or method and cannot be declared as final"
+    end
+    if final_module?(mod)
+      raise "#{mod.name} was already declared as final and cannot be re-declared as final"
+    end
+    if T::AbstractUtils.abstract_module?(mod)
+      raise "#{mod.name} was already declared as abstract and cannot be declared as final"
+    end
+    if mod.is_a?(Class)
+      orig_inherited = T::Private::ClassUtils.replace_method(mod.singleton_class, :inherited) do |arg|
```

Comment:
Should a final `module` be not `include` / `extend`-able?

Reply:
Hm. Maybe we do want that. I mentioned earlier how making a regular non-class module final doesn't do anything for you other than make all the methods final, but that might be a good idea.